apps/builder/app/canvas/features/webstudio-component/webstudio-component.tsx

```diff
@@ -108,12 +119,81 @@ type WebstudioComponentDevProps = {
   components: Components;
 };
 
-export const WebstudioComponentDev = ({
-  instance,
-  instanceSelector,
-  children,
-  components,
-}: WebstudioComponentDevProps) => {
+/**
+ * Radix's VisuallyHiddenPrimitive.Root https://github.com/radix-ui/primitives/blob/main/packages/react/visually-hidden/src/VisuallyHidden.tsx
+ * component makes content from hidden elements accessible to screen readers.
+ * Using the same Content children, however, leads to duplicated React elements, breaking our 'isSelected' logic.
+ * To prevent this, we check if an instance is a descendant of VisuallyHiddenPrimitive.Root, and if so,
+ * we avoid rendering it.
+ */
+const useIsScreenReaderDescendant = (ref: RefObject<HTMLElement>) => {
+  const [isScreenReaderDescendant, setIsScreenReaderDescendant] =
+    useState(false);
+
+  useEffect(() => {
+    if (ref.current !== null) {
+      for (
+        let element: HTMLElement | null = ref.current;
+        element !== null;
+        element = element.parentElement
+      ) {
+        if (
+          element.style.overflow === "hidden" &&
```

Comment:
probably should compute the style, because it could have been applied via css as welll, if we want it to work universally, not just with radix

Reply:
can this condition be more universal in general, not just targeted for radix? Effectively you are trying to detect visually hidden element, right?

could it be visually hidden by clip and overflow hidden without position absolute and width/height?

Reply:
react-aria has near same a little bit other set of props in `style` attribute. 

But overflow, height, width, position, and clip are common.

aria example:
```
border: 0px; clip: rect(0px, 0px, 0px, 0px); clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;
```

Also using the `style` attribute intead of getComputedStyle make it less possible to interfere with user defined styles as in our system they are all classes.

What if real user will create visually hidden box for its own needs. We should not skip user defined elements.

Reply:
in this case maybe the approach is not great and we should search for a way to explicitely tell builder from component that this component has a visually hidden thing? e.g. a hint using a special attribute or something?

Reply:
Wouldn't somehow improve things, nothing would prevent user to make such element inside tooltip content.
Im not even sure that react-aria adds VisuallyHidden elements implicitly like radix.
And that this is issue is common across most frameworks/components